pkg/storaged/fsys-tab.jsx

```diff
@@ -137,14 +154,12 @@ export function mounting_dialog(client, block, mode) {
         var new_config = null;
         var all_new_opts;
 
-        if (old_config) {
```

Comment:
This seems unrelated and possibly wrong? (not sure though why we needed the check in the first place)

Reply:
This is indeed non-obvious, but I'd say the code was buggy earlier.  With the check, we could not set any options when mounting a filesystem that wasn't listed in /etc/fstab.

When adding an entry to /etcfstab with this function (as opposed to changing an existing one), it would always end up having "default" options. Now we want to add "noauto" in certain cases, and that's how I noticed.

I guess I should add a test for this...

Reply:
Done.